### PR TITLE
[many ports]switch to vcpkg-cmake / vckg-cmake-config part 1

### PR DESCRIPTION
**Pull request by JackBoosY**
- Switch `vcpkg_fixup_cmake_targets` to `vcpkg_cmake_config_fixup`
- Switch `vcpkg_*_cmake` to `vcpkg_cmake_*`
- Add double quotes to the absolute paths.

### CODE REVIEW

**Comment by JackBoosY** [comment]
There may have 400+ files so I split them into 4 PRs.

**Comment by autoantwort** [comment]
Have you used a script for it like in https://github.com/microsoft/vcpkg/pull/26981?

**Comment by JackBoosY** [comment]
> Have you used a script for it like in #26981?

No, completely manual.

**Comment by JackBoosY** [comment]
@autoantwort BTW, you should keep the number of modified files in your PR to a relatively small number.

**Comment by JackBoosY** [comment]
arpack-ng regression is a baseline issue.

**Comment by JackBoosY** [comment]
Depends on https://github.com/microsoft/vcpkg/pull/29935

**Comment by JackBoosY** [comment]
@jimwang118 request review.

File: ports/ade/portfile.cmake [hunk 1]
```diff
@@ -6,19 +6,18 @@ vcpkg_from_github(
...
+vcpkg_cmake_config_fixup()
 
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
 
 # Handle copyright
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/ade RENAME copyright)
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (jimwang118) [inline ports/ade/portfile.cmake:23]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

File: ports/ade/portfile.cmake (hunk 1 above)
Comment (JackBoosY) [reply ports/ade/portfile.cmake:23]: No no, the number to replace the new function `vcpkg_install_copyright` is huge, and it's not the target of this PR either.

Comment (jimwang118) [reply ports/ade/portfile.cmake:23]: Ok，then fix this in another PR.

**Review by jimwang118 — APPROVED** [review]

**Pull request merged**